Daily DCU test failed since 20241102

### Describe the bug

Since 20241102, the DCU test is failed because of compile error. This error is introduced by commits between 0f44046 (tested at 20241102, and is failed) and 3984296 (tested at 20241101, and is passed).

The error is related to force/stress, which may be introduced by commit e276480937.

@dyzheng , could you have a check?

![Image](https://github.com/user-attachments/assets/6deb7331-9e0d-4345-acb8-ff4aabde2144)


The error message is:
```
/public/home/abacus/abacus-dcu/source/module_hamilt_pw/hamilt_pwdft/kernels/rocm/stress_op.hip.cu:294:5: error: use of undeclared identifier '__syncwarp'; did you mean '__sync_swap'?
    __syncwarp();
    ^~~~~~~~~~
    __sync_swap
/public/home/abacus/abacus-dcu/source/module_hamilt_pw/hamilt_pwdft/kernels/rocm/stress_op.hip.cu:294:5: note: '__sync_swap' declared here
/public/home/abacus/abacus-dcu/source/module_hamilt_pw/hamilt_pwdft/kernels/rocm/stress_op.hip.cu:294:16: error: too few arguments to function call, expected at least 1, have 0
    __syncwarp();
    ~~~~~~~~~~ ^
[ 70%] Building CXX object source/module_io/CMakeFiles/io_basic.dir/read_wfc_to_rho.cpp.o
2 errors generated when compiling for gfx906.
[ 70%] Building CXX object source/module_cell/CMakeFiles/cell.dir/read_pp_vwr.cpp.o
CMake Error at device_rocm_generated_stress_op.hip.cu.o.cmake:180 (message):
  Error generating file
  /public/home/abacus/abacus-dcu/build-dcu/source/CMakeFiles/device_rocm.dir/module_hamilt_pw/hamilt_pwdft/kernels/rocm/./device_rocm_generated_stress_op.hip.cu.o


make[2]: *** [source/CMakeFiles/device_rocm.dir/build.make:147: source/CMakeFiles/device_rocm.dir/module_hamilt_pw/hamilt_pwdft/kernels/rocm/device_rocm_generated_stress_op.hip.cu.o] Error 1
make[2]: *** Waiting for unfinished jobs....
```

### Expected behavior

_No response_

### To Reproduce

_No response_

### Environment

_No response_

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [ ] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [ ] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [ ] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

--- a/source/module_hamilt_pw/hamilt_pwdft/kernels/rocm/stress_op.hip.cu
+++ b/source/module_hamilt_pw/hamilt_pwdft/kernels/rocm/stress_op.hip.cu
@@ -291,7 +291,6 @@ __global__ void cal_stress_nl(
         ++iat;
         sum+=nproj;
     }//ia
-    __syncwarp();
     warp_reduce(stress_var);
     if (threadIdx.x % WARP_SIZE == 0) {
         atomicAdd(stress + ipol * 3 + jpol, stress_var);

--- a/source/module_hamilt_pw/hamilt_stodft/kernels/rocm/hpsi_norm_op.hip.cu
+++ b/source/module_hamilt_pw/hamilt_stodft/kernels/rocm/hpsi_norm_op.hip.cu
@@ -45,7 +45,7 @@ void hamilt::hpsi_norm_op<FPTYPE, base_device::DEVICE_GPU>::operator()(const bas
       npwk_max, npwk, Ebar, DeltaE,
       reinterpret_cast<thrust::complex<FPTYPE>*>(hpsi),
       reinterpret_cast<const thrust::complex<FPTYPE>*>(psi_in));
-    cudaCheckOnDebug();
+    hipCheckOnDebug();
 }
 
 template struct hpsi_norm_op<float, base_device::DEVICE_GPU>;

